### PR TITLE
Hdf5 library

### PR DESCRIPTION
**Pull request by jontromanab**
This PR addresses issue #13 
All the scopes where the data is stored in .h5 file or loaded from .h5 file, can now use the hdf5_dataset class with proper arguments.
All the files using hdf5 are changed accordingly.
As the new MultiMap structure uses pointer, loading the data again into MultiMap does not maintain the same address structure. It is been discussed in detail in hdf5_dataset.cpp line 411

@akgoins Please review this PR and let me know with issues

### CODE REVIEW

File: map_creator/include/map_creator/hdf5_dataset.h [hunk 1]
```diff
@@ -5,20 +5,64 @@
...
+  bool h5ToResolution(float &resolution);//Accesses the resolution of the map
+
+private:
+
+  bool h5ToVectorCap(Vector &capabilityData); //Accesses the capability data
+  bool h5ToMultiMapPosesAndSpheres(MultiMap  &PoseCol, Map &SphereCol); //loads the whole data with same address structure as stored in .h5
+
+  bool h5ToMultiMapPoses(multiMap &PoseCol, map_D &SphereCol); //accesses the poses and spheres data in the poses dataset
+  bool h5ToMultiMapPoses(multiMap &PoseCol); //Accessess only the data from poses dataset regardless of address
+  bool h5ToMultiMapSpheres(map_D& SphereCol); //Accessess only the data from spheres dataset regardless of address
```
Comment (akgoins) [inline map_creator/include/map_creator/hdf5_dataset.h]: All variables need to be lower_case_under_score (Poses, Spheres, PoseCol, SphereCol, etc.).

File: map_creator/src/create_reachability_map.cpp [hunk 2]
```diff
@@ -189,243 +188,16 @@ int main(int argc, char **argv)
...
-
-      poseReach.push_back(poseAndSphere);
     }
 
     ROS_INFO("No of spheres reachable: %lu", sphereColor.size());
 
     // Creating maps now
+//Saving map to dataset
+    hdf5_dataset::Hdf5Dataset h5(filename);
+    h5.saveReachMapsToDataset(PoseColFilter, sphereColor, resolution);
```
Comment (akgoins) [inline map_creator/src/create_reachability_map.cpp]: variable names need to be lower_case_under_scored.

File: map_creator/src/load_capability_map.cpp [hunk 3]
```diff
@@ -26,58 +18,38 @@ int main(int argc, char **argv)
...
-    herr_t ret = H5Aread(attr, H5T_NATIVE_FLOAT, &res);
+    hdf5_dataset::Hdf5Dataset h5(argv[1]);
+    h5.open_cap();
 
-    // Closing resources
-    H5Aclose(attr);
-    H5Dclose(cap_dataset);
-    H5Gclose(cap_group);
-    H5Fclose(file);
+    std::vector<std::vector<double> > capData;
```
Comment (akgoins) [inline map_creator/src/load_capability_map.cpp]: variable name lower_case_under_scored.

File: map_creator/src/load_reachability_map.cpp [hunk 4]
```diff
@@ -33,71 +24,42 @@ int main(int argc, char **argv)
...
 
-    // Sphere dataset
-    sphere_group = H5Gopen(file, SPHERE_GROUPNAME, H5P_DEFAULT);
-    sphere_dataset = H5Dopen(sphere_group, SPHERE_DATASETNAME, H5P_DEFAULT);
-
-    std::multimap< std::vector< double >, double > SphereCol;
-    hd5.h5ToMultiMapSpheres(sphere_dataset, SphereCol);
-
-    // Resolution Attribute
+    MultiMap PoseColFilter;
```
Comment (akgoins) [inline map_creator/src/load_reachability_map.cpp]: Variable naming.

Comment (jontromanab) [inline map_creator/src/load_reachability_map.cpp]: @akgoins All the namespaces are corrected and CamelCased
variables are lower_case_under_scored
SpereCol is resized is base_placement_widget

Please check again and merge the PR

**Comment by akgoins** [comment]
+1.

**Pull request merged**